go/adbc/driver/databricks/record_builder.go

```diff
@@ -0,0 +1,338 @@
+package databricks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+)
+
+// ColumnBuilder is an interface for building Arrow arrays
+type ColumnBuilder interface {
+	Append(value interface{}) error
```

Comment:
What is this type?

Reply:
`interface` is equivalent to python's `any` type hint

Reply:
Crazy because it's unchecked? Note that it's checked when the builder implementation is instantiated based on the schema.